File: backend/src/main/kotlin/com/project/advanced/project/domain/valueobject/EstadoReserva.java
package com.project.advanced.project.domain.valueobject

public enum class EstadoReserva {
    CANCELADA(false),
    CONFIRMADA(true),
    EN_CURSO(true),
    FINALIZADA(false),
    NO_SHOW(false),
    PENDIENTE(true);

    private final boolean esActiva;

    EstadoReserva(boolean esActiva) {
      this.esActiva = esActiva;
    }

    public boolean esActiva() {
      return esActiva;
    }

    public boolean esTerminal() {
      return !esActiva;
    }
}

File: backend/src/main/kotlin/com/project/advanced/project/domain/valueobject/EstadoReserva.kt
package com.project.advanced.project.domain.valueobject

enum class EstadoReserva(
    private val esActiva: Boolean,
) {
    CANCELADA(false),
    CONFIRMADA(true),
    EN_CURSO(true),
    FINALIZADA(false),
    NO_SHOW(false),
    PENDIENTE(true),
    ;

    fun esActiva(): Boolean = esActiva

    fun esTerminal(): Boolean = !esActiva
}
